### PR TITLE
V1.2.1

### PR DESCRIPTION
**Pull request by Aran30**
Added template dockerfile

Updated core to version 1.2.1

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -4,8 +4,9 @@
 name: Java CI with Gradle
 
 # Triggers the workflow on push or pull request events (on every branch)
-on: [push, pull_request]
-
+on:
```
Comment (lakhoune) [inline .github/workflows/gradle.yml:7]: why did you remove the pull_request hook? Also, the v1.2.1 branch can be removed.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+FROM openjdk:17-jdk-alpine
+
+ENV HTTP_PORT=8080
+ENV HTTPS_PORT=8443
+ENV LAS2PEER_PORT=9011
+
+RUN apk add --update bash mysql-client apache-ant tzdata curl && rm -f /var/cache/apk/*
```
Comment (lakhoune) [inline Dockerfile:7]: apache-ant and MySQL can be removed

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    adduser -u 1000 -S las2peer -G las2peer
+
+COPY --chown=las2peer:las2peer . /src
+WORKDIR /src
+
+# run the rest as unprivileged user
+USER las2peer
+RUN chmod -R a+rwx /src
+RUN chmod +x /src/docker-entrypoint.sh
+RUN chmod +x gradlew && ./gradlew build --exclude-task test
```
Comment (lakhoune) [inline Dockerfile:18]: why is the  test task excluded?

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -e
+
+# print all comands to console if DEBUG is set
+if [[ ! -z "${DEBUG}" ]]; then
+    set -x
+fi
+NODE_ID_SEED=${NODE_ID_SEED:-$RANDOM}
```
Comment (lakhoune) [inline docker-entrypoint.sh:9]: Why is this line actually needed?

File: docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+export WEB_CONNECTOR_PROPERTY_FILE='etc/i5.las2peer.connectors.webConnector.WebConnector.properties'
+export SERVICE_VERSION=$(awk -F "=" '/service.version/ {print $2}' gradle.properties)
+export SERVICE_NAME=$(awk -F "=" '/service.name/ {print $2}' gradle.properties)
+export SERVICE_CLASS=$(awk -F "=" '/service.class/ {print $2}' gradle.properties)
+export SERVICE=${SERVICE_NAME}.${SERVICE_CLASS}@${SERVICE_VERSION}
+
+function set_in_service_config {
+    sed -i "s?${1}[[:blank:]]*=.*?${1}=${2}?g" ${SERVICE_PROPERTY_FILE}
+}
+#cp $SERVICE_PROPERTY_FILE $SERVICE_PROPERTY_FILE
```
Comment (lakhoune) [inline docker-entrypoint.sh:22]: what is this line for?

**Review by lakhoune — CHANGES_REQUESTED** [review]

File: docker-entrypoint.sh (hunk 4 above)
Comment (Aran30) [reply docker-entrypoint.sh:9]: a node_id_seed always needs to be defined when starting a l2p node whether it is random or not, as based on that seed, the running node id will be generated

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (Aran30) [reply .github/workflows/gradle.yml:7]: they didn't work

File: Dockerfile (hunk 2 above)
Comment (Aran30) [reply Dockerfile]: done

File: docker-entrypoint.sh (hunk 5 above)
Comment (Aran30) [reply docker-entrypoint.sh:22]: which line do you mean?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+RUN chmod +x gradlew && ./gradlew build 
+#RUN dos2unix /src/docker-entrypoint.sh
+#RUN dos2unix /src/etc/i5.las2peer.connectors.webConnector.WebConnector.properties
+#RUN dos2unix /src/etc/i5.las2peer.services.servicePackage.akgService.properties
+
+EXPOSE $HTTP_PORT
+EXPOSE $HTTPS_PORT
+EXPOSE $LAS2PEER_PORT
+RUN chmod +x /src/docker-entrypoint.sh
+RUN chmod +x docker-entrypoint.sh
```
Comment (lakhoune) [inline Dockerfile:27]: Either this line or the one above is not needed

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+WORKDIR /src
+
+# run the rest as unprivileged user
+USER las2peer
+RUN chmod -R a+rwx /src
+RUN chmod +x /src/docker-entrypoint.sh
+RUN chmod +x gradlew && ./gradlew build 
+#RUN dos2unix /src/docker-entrypoint.sh
+#RUN dos2unix /src/etc/i5.las2peer.connectors.webConnector.WebConnector.properties
+#RUN dos2unix /src/etc/i5.las2peer.services.servicePackage.akgService.properties
```
Comment (lakhoune) [inline Dockerfile:21]: Maybe add a comment here telling users why this is commented out

**Review by lakhoune — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 6 above)
Comment (Aran30) [reply Dockerfile:27]: removed it now

**Review by lakhoune — APPROVED** [review]
